SmartDeviceLink/SDLStreamingMediaConfiguration.h

```diff
@@ -47,7 +47,20 @@ NS_ASSUME_NONNULL_BEGIN
 
  @warning This is a weak property and it's therefore your job to hold a strong reference to this window.
  */
-@property (weak, nonatomic, nullable) UIWindow *window;
+@property (weak, nonatomic, nullable) UIWindow *window __deprecated_msg("Use rootViewController instead");
+
+/**
+ Set the initial view controller your video streaming content is within.
+
+ Activates the haptic view parser and CarWindow systems when set. This library will also use that information to attempt to return the touched view to you in `SDLTouchManagerDelegate`.
+
+ @note If you wish to alter this rootViewController while streaming via CarWindow, you must set a new `rootViewController` on `SDLStreamingMediaManager` and this will update both the haptic view parser and CarWindow.
+
+ @warning Apps using views outside of the `UIView` heirarchy (such as OpenGL) are currently unsupported. If you app uses partial views in the heirarchy, only those views will be discovered. Your OpenGL views will not be discoverable to a haptic interface head unit and you will have to manually make these views discoverable via the `SDLSendHapticData` RPC request.
+
+ @warning This is a weak property and it's therefore your job to hold a strong reference to this view controller.
+ */
+@property (strong, nonatomic, nullable) UIViewController *rootViewController;
```

Comment:
Why are we forcing user's to init a `rootViewController` in the `SDLStreamingMediaConfiguration`? Since it is getting initialized in the `AppDelegate` before the phone's `UIWindow` `rootViewController` is setup, I am forced to set a dummy `rootViewController ` and then set the actual view controller that I want to stream later.

Reply:
After investigating about making it friendlier, it is not possible without a major version change due to the need to continue to support the `window` property in `SDLStreamingMediaConfiguration`.